### PR TITLE
Add log4j2 integration

### PR DESCRIPTION
**Pull request by RecuencoJones**
Fixing issues with #41

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -35,12 +35,18 @@ repositories {
 
 configurations {
     providedRuntime
+    all {
+        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-logging'
+    }
 }
 
 dependencies {
+
```
Comment (SAReyes) [inline build.gradle]: Unneeded line, please keep the consistency in the whole file.

File: src/main/java/es/unizar/webeng/hello/CounterHandler.java [hunk 2]
```diff
@@ -23,7 +24,9 @@
 @Component
 public class CounterHandler extends TextWebSocketHandler {
 
-    private static final Logger logger = Logger.getLogger(CounterHandler.class);    // log4j logger
+
```
Comment (SAReyes) [inline src/main/java/es/unizar/webeng/hello/CounterHandler.java]: Remove this line

File: src/main/java/es/unizar/webeng/hello/CounterHandler.java [hunk 3]
```diff
@@ -23,7 +24,9 @@
...
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
```
Comment (SAReyes) [inline src/main/java/es/unizar/webeng/hello/CounterHandler.java]: Remove this line

File: src/main/java/es/unizar/webeng/hello/CounterHandler.java [hunk 4]
```diff
@@ -23,7 +24,9 @@
 @Component
 public class CounterHandler extends TextWebSocketHandler {
 
-    private static final Logger logger = Logger.getLogger(CounterHandler.class);    // log4j logger
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (SAReyes) [inline src/main/java/es/unizar/webeng/hello/CounterHandler.java]: Stick with static loggers please, we've been using those from previous versions.

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,6 +1,6 @@
-#Thu Sep 22 22:20:06 CEST 2016
+#Thu Sep 29 00:15:44 CEST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.13-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.13-all.zip
```
Comment (SAReyes) [inline gradle/wrapper/gradle-wrapper.properties]: Is it required to modify this? @fjlopez, how does this affect the app? can you review it please?

File: build.gradle [hunk 6]
```diff
@@ -35,12 +35,18 @@ repositories {
 
 configurations {
     providedRuntime
+    all {
```
Comment (RecuencoJones) [inline build.gradle]: is the logger lib getting excluded for every build configuration? does this mean we are not logging anything even in development mode?

File: src/main/java/es/unizar/webeng/hello/CounterHandler.java (hunk 4 above)
Comment (adrianblade) [inline src/main/java/es/unizar/webeng/hello/CounterHandler.java]: is correct we want a static log instance

File: gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (adrianblade) [inline gradle/wrapper/gradle-wrapper.properties]: It's the same version but the zip  file add extra content
BIN
- bin
-init.d
-lib
-media
ALL
- bin
-init.d
-lib
-media
-docs
-samples
-src

the core file are the same

**Comment by nebur395** [comment]
Do I have to do a new pull request? I have the changes in my remote.

**Comment by RecuencoJones** [comment]
@nebur395 
> Sorry for all the confusion, @RecuencoJones will take care of this PR.

**Comment by fjlopez** [comment]
I think that the best solution for these cases is that only **me** should decide the merge. Too many cooks may spoil the dish.

File: gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (fjlopez) [inline gradle/wrapper/gradle-wrapper.properties]: Yep, bin is smaller. So revert to the previous version..

File: src/main/java/es/unizar/webeng/hello/CounterHandler.java (hunk 4 above)
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/CounterHandler.java]: You may keep `logger` in lowercase as it is not a constant. See [this discussion](http://stackoverflow.com/questions/1417190/should-a-static-final-logger-be-declared-in-upper-case).

File: gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (RecuencoJones) [inline gradle/wrapper/gradle-wrapper.properties]: Done

File: src/main/java/es/unizar/webeng/hello/CounterHandler.java (hunk 4 above)
Comment (RecuencoJones) [inline src/main/java/es/unizar/webeng/hello/CounterHandler.java]: It is already in lower case

**Comment by fjlopez** [comment]
I merge the change. @nebur395 will fix the issues in his next pull request.

**Pull request merged**